### PR TITLE
Fix one likely error with a collection key.

### PR DESCRIPTION
**Pull request by wmdietl**
Suppress other warnings by casting to the expected type.

### CODE REVIEW

File: framework/src/org/checkerframework/framework/util/typeinference/solver/ConstraintMap.java [hunk 1]
```diff
@@ -178,7 +178,8 @@ public void addTypeSubtype(
             final AnnotatedTypeMirror supertype,
             AnnotationMirrorSet hierarchies) {
         final Subtypes subtypes = targetToRecords.get(target).subtypes;
-        final AnnotationMirrorSet subtypesTops = subtypes.targets.get(supertype);
+        final AnnotationMirrorSet subtypesTops =
+                subtypes.targets.get(supertype.getUnderlyingType());
```
Comment (wmdietl) [inline framework/src/org/checkerframework/framework/util/typeinference/solver/ConstraintMap.java]: Without the `getUnderlyingType` the `get` will always return null. Is this the right fix?

Comment (smillst) [reply framework/src/org/checkerframework/framework/util/typeinference/solver/ConstraintMap.java]: No, I think it should be `subtypes.types.get(supertype)`.

Comment (wmdietl) [reply framework/src/org/checkerframework/framework/util/typeinference/solver/ConstraintMap.java]: I tried changing the code as you suggested, but then I get failures for `all-systems/java8/memberref` tests: `    [junit] MemberReferences.java:34: warning: (methodref.inference.unimplemented)`
As all tests seem to pass with the current changes and this code will be refactored soon, should we keep this? Or should I replace it with `subtypesTops = null` which might be equivalent?

Comment (smillst) [reply framework/src/org/checkerframework/framework/util/typeinference/solver/ConstraintMap.java]: That seems to be the expected warning:

```
 32             // TODO: Issue 802: type argument inference
 33             // TODO: should expect warning: (methodref.inference.unimplemented)
 34             FunctionMR f2 = super::func2;
```

Comment (smillst) [reply framework/src/org/checkerframework/framework/util/typeinference/solver/ConstraintMap.java]: It appears that is flakey test:

https://github.com/typetools/checker-framework/pull/1480#discussion_r135657561

Comment (wmdietl) [reply framework/src/org/checkerframework/framework/util/typeinference/solver/ConstraintMap.java]: Even on master, when running `ant aliasing-tests` in `framework` I get:

```
    [junit] Testsuite: tests.AliasingTest
    [junit] Tests run: 6, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 2.361 sec
    [junit] 
    [junit] Testcase: run[all-systems](tests.AliasingTest):     FAILED
    [junit] 0 out of 4 expected diagnostics were found.
    [junit] 4 expected diagnostics were not found:
    [junit] RawTypeAssignment.java:21: warning: [unchecked] unchecked conversion
    [junit] GenericsCasts.java:8: warning: [unchecked] unchecked cast
    [junit] GenericsCasts.java:22: warning: [unchecked] unchecked cast
    [junit] GenericsCasts.java:42: warning: [unchecked] unchecked cast
    [junit] 2 unexpected diagnostics were found:
    [junit] EqualityTests.java:1: error: package org.checkerframework.checker.interning.qual does not exist
    [junit] EqualityTests.java:4: error: cannot find symbol
```

The errors in EqualityTests.java seem reasonable: framework doesn't know about interned.
This commit https://github.com/typetools/checker-framework/commit/8a5cba5f9342c86ddcabc2ee4eb4195573f80f78#diff-97851f401e871eef809ae68fa74beb66
disregarded the existing comment and introduced `@Interned` annotations in a framework test.
@mernst why did you perform that change?

We should also look through our travis setup to make sure that we execute the `framework` tests independently of `checker`.

I'm not sure about the unchecked cast warnings.

**Pull request merged**